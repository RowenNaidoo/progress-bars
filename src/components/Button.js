import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Button = ({ value, buttonClicked }) => (
  <StyledButton onClick={() => buttonClicked(value)}>{value}</StyledButton>
)

const StyledButton = styled.button`
  flex: 1 1 auto;
  margin: 5px 0;
  height: 40px;
  transition: all 0.4s ease 0s;
  font-weight: bold;
  background-color: #85929E;

  @media (min-width: 576px) {
    flex: 1 1 auto;
    margin: 0 5px;
  }

  &:hover {
    cursor: pointer;
    background-color: #34495E;
    border: 1px solid #34495E;
    transition: all 0.4s ease 0s;
    color: white;
  }
`

Button.propTypes = {
  value: PropTypes.number,
  buttonClicked: PropTypes.func
}

export default Button