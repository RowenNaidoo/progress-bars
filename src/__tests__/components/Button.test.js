import React from 'react'
import { shallow } from 'enzyme'
import Button from '../../components/Button'


describe('#Button', () => {

  test('renders learn react link', () => {
    const onButtonClicked = jest.fn()
    const component = shallow(<Button value={50} buttonClicked={onButtonClicked} />)
    expect(component).toMatchSnapshot()
  })

})