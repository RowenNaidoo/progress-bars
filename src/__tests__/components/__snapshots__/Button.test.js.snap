// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Button renders learn react link 1`] = `ShallowWrapper {}`;
